sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -436,3 +436,21 @@ case class Murmur3Hash(children: Seq[Expression], seed: Int) extends Expression
     }
   }
 }
+
+/**
+  * An expression that will print the value of child to stderr (used for debugging codegen).
+  */
+case class Echo(child: Expression) extends UnaryExpression {
```

Comment:
This could be used to print out the the result of any expression, for example
```
  Sqrt(Echo(a * b))
```
You can see the result of `a * b`